File: test/fixtures/dangling/ios/Sources/DanglingPlugin.swift
import Foundation
import Capacitor

@objc(DanglingPlugin)
public class DanglingPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "DanglingPlugin"
    public let jsName = "Dangling"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "ping", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "fetch", returnType: CAPPluginReturnPromise)
    ]

    @objc func ping(_ call: CAPPluginCall) {
        call.resolve(["alive": true])
    }

    @objc func fetch(_ call: CAPPluginCall) {
        call.resolve()
    }
}

File: test/fixtures/dangling/android/src/main/java/com/example/dangling/DanglingPlugin.java
package com.example.dangling;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Dangling")
public class DanglingPlugin extends Plugin {

    @PluginMethod
    public void ping(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("alive", true);
        call.resolve(ret);
    }

    @PluginMethod
    public void fetch(PluginCall call) {
        call.resolve();
    }
}
